wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/ResourceLimitAttributesProvisioner.java

```diff
@@ -71,10 +71,11 @@ public static void provisionMemory(
     final Map<String, String> attributes = machineConfig.getAttributes();
     String configuredLimit = attributes.get(MEMORY_LIMIT_ATTRIBUTE);
     String configuredRequest = attributes.get(MEMORY_REQUEST_ATTRIBUTE);
-    if (isNullOrEmpty(configuredLimit)) {
+    // Added < 0  check to avoid bypassing limit if someone is set negative value into attribute.
+    if (isNullOrEmpty(configuredLimit) || Long.parseLong(configuredLimit) <= 0) {
```

Comment:
is `configuredLimit` and `configuredRequest` already converted to bytes so it can be long parsed ? I assume so, just making sure...

Reply:
sorry, did not got the question..

Reply:
you can set limit/request in "human readable" form, like `100m`, which can't be directly parsed into long. My question is, whether you are sure here, that `configuredLimit` is no longer original human readable format, but is converted into raw bytes, that can be parsed as long.

Reply:
Oh yes. Internally it's always bytes or core fractions.